src/exporters/dir.js

```diff
@@ -14,15 +15,17 @@ module.exports = (node, name, dagService) => {
       path: path.join(name, link.name),
       hash: link.hash
     })),
-    pull.map((item) => pull(
-      dagService.getStream(item.hash),
-      pull.map((n) => switchType(
+    paramap((item, cb) => dagService.get(item.hash, (err, n) => {
+      if (err) {
+        return cb(err)
+      }
+
+      cb(null, switchType(
         n,
         () => pull.values([item]),
         () => fileExporter(n, item.path, dagService)
-      )),
-      pull.flatten()
-    )),
+      ))
+    })),
```

Comment:
@dignifiedquire could you please add a bit of documentation to this function, each time I come to read it, I scratch my head to get back to understand what is supposed to be going on